### ISSUE TITLE
config does not play well with node-support-source-map

### ISSUE THREAD

**Issue by PavelPolyakov** [node-config/node-config#530]
### I'm submitting a ... 
  - [x] bug report

### What is the current behavior?
Currently, if you store your configs using `ts` extension and load them using `config` - `source-map-support` for node applications stops working. By not working I mean - there are no references to the original `ts` files in the stack trace of the errors.

I consider this functionality very important.

![image](https://user-images.githubusercontent.com/839290/51191096-fa731680-18e3-11e9-9dee-fb16d14f384a.png)

### What is the expected behaviour?
They can work together and don't interfere each other (namely, `config` has no affect on how other modules work).

Expected behaviour:
![image](https://user-images.githubusercontent.com/839290/51191112-04951500-18e4-11e9-8116-615119498637.png)

### Please tell us about your environment:

 -  node-config version: v3
 -  node-version: v10
 
### Other information

Here is the repo where you can reproduce the issue:
https://github.com/PavelPolyakov/config-vs-node-source-map-support

**Comment by lorenwest** [comment]
I have a hard time categorizing this as a bug. There is no documented expectation of this library working well with this other library. 

This sounds more like a feature request than a bug report. Anyone with .ts knowledge want to make a contribution for this issue?

**Comment by lorenwest** [comment]
Agreed - it would be a good addition/improvement. 

We have a few contributors that have added .ts support, and I'll mark this issue as a collaboration request.

**Comment by mrvisser** [comment]
I've updated the wiki (because it actually let me...) to indicate that `.ts` config files will mysteriously break source maps:

https://github.com/lorenwest/node-config/wiki/Configuration-Files#typescript---ts

This issue is devious enough that I think it warrants it.

**Comment by andrzej-woof** [comment]
Hey, I've submitted PR https://github.com/node-config/node-config/pull/721 with a possible fix.

Tested locally on the demo repo https://github.com/PavelPolyakov/config-vs-node-source-map-support (after updating deps) and it seems to work on my end

**Comment by andrzej-woof** [comment]
Also I think it might improve or even fix https://github.com/node-config/node-config/issues/613

**Comment by EvanTedesco** [comment]
I tried using @andrzej-woof's branch and I still had incorrect line reporting for failing Jest tests.  The PR [here](https://github.com/node-config/node-config/pull/724) seemed to fix the issue for me.  require.extensions is deprecated and the require.extensions object does not even have the .ts key(It only has  .js, .json, and .node as far as I can tell) so I am not sure the original intent of the removed code.  In any case this seems to fix the incorrect line reporting issue on Jest test failures for my project so I figured I would test Cunningham's law and make a PR in hopes that someone more familiar with this library would save me a bit of code archeology.  

Node v16.13.1
node-config: v3.3.9
ts-node: v10.2.0
ts-jest: v28.0.8

**Comment by andrzej-woof** [comment]
Hi @EvanTedesco do you run your code via ts-node/ts-jest? 
If so your issue is a slightly different one. The original problem in the repo concerns running tsc compiled code directly in node and transforming stack traces from js back to ts files with source maps.

I admit my PR won't solve your problem (as far as I understand it), but by removing this code I think you effectively disable loading any ts config files with compiled code (unsure if they're loaded at all even with ts-node. 

Of course I might be wrong, just sharing thoughts here

**Comment by andrzej-woof** [comment]
A bit more context, after some further investigation, regarding `require.extensions` being deprecated see discussion [here](https://github.com/nodejs/node/issues/32483) - there seems to be on alternatives provided, and that property is still wildely used by other tools ([babel](https://github.com/babel/babel/blob/76886ac50cd6afacadbe09eaaa457236a8c8f8cd/packages/babel-core/src/config/files/module-types.ts#L58) for example)

As for running with `ts-node` it seems it works correctly on https://github.com/PavelPolyakov/config-vs-node-source-map-support when with `ts-node ./src/server.ts` (version v10.9.1), `require.extensions` is set, stack trace points to `ts` files. I have not checked `ts-jest`

**Comment by EvanTedesco** [comment]
> Hi @EvanTedesco do you run your code via ts-node/ts-jest? If so your issue is a slightly different one. The original problem in the repo concerns running tsc compiled code directly in node and transforming stack traces from js back to ts files with source maps.
> 
> I admit my PR won't solve your problem (as far as I understand it), but by removing this code I think you effectively disable loading any ts config files with compiled code (unsure if they're loaded at all even with ts-node.
> 
> Of course I might be wrong, just sharing thoughts here

Yes I am running with ts-node/ts-jest.  Thank you for the elaboration/clarification.

**Comment by jdmarshall** [comment]
This bug is a consequence of #787 and also breaks nyc if you use TS config files, as I discovered when I added code coverage to node-config.
